### PR TITLE
Feature PeerService

### PR DESCRIPTION
**Pull request by satellitex**
# What is this pull request?

Write implement interface about peer-service.
But, SelfStatus is temporary implement.

# Why do you implement it? Why do we need this pull request?

PeerService  must be used by iroha.

### CODE REVIEW

**Comment by Warchant** [comment]
I have a feeling that peer service should store this info about peers:

```
struct NetworkNode {
    std::string ip;
    uint16_t port;
    ed25519::pubkey_t pubkey;
  };

  struct PeerService {
    std::vector<NetworkNode> peers;
    ...
```

**Comment by satellitex** [comment]
@Warchant 
I thought `ip` is included by `port`. (e.g: ip = "0.0.0.0:8080")
If we need `uint16_t port` in another, we should add `port` to `model::peer` and `protocol::peer`.

File: irohad/consensus/sumeragi.cpp [hunk 1]
```diff
@@ -56,18 +54,22 @@ namespace consensus {
     }
 
     size_t getMaxFaulty() {
-      return (size_t) peer_service::monitor::getActivePeerSize() / 3;
+      return 1;
+//      return (size_t) peer_service::monitor::getActivePeerSize() / 3;
```
Comment (takemiyamakoto) [inline irohad/consensus/sumeragi.cpp]: "TODO:"を追加したら状態をわかりやすくなります

File: irohad/peer_service/peer_service.cpp [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    /**
+     * @return List of peers that therefore permutation.
+     */
+    std::vector<std::shared_ptr<Peer>> getPermutationPeers() { return std::vector<std::shared_ptr<Peer>>(); }
+
+    /**
+     * @param i index
+     * @return A i-th peer that therefore permutation.
+     */
+    Peer getPermutationAt(int i) { return Peer(); }
```
Comment (takemiyamakoto) [inline irohad/peer_service/peer_service.cpp]: これは途中？途中なら、TODO:を追加するとわかりやすくなります。

File: irohad/peer_service/peer_service.cpp [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+     return detail::self_;
+    }
+
+    /**
+     * When on_porposal sends on_commit, it is called.
+     * It check signs from Block, and identify dead peers which It throw to
+     * issue Peer::Remove transaction.
+     * @param commited_block commited block with signs
+     */
+    void DiesRemove(const iroha::model::Block &commited_block) {}
```
Comment (takemiyamakoto) [inline irohad/peer_service/peer_service.cpp]: *RemoveDeadPeer* is a better name.

File: irohad/peer_service/peer_service.hpp [hunk 4]
```diff
@@ -18,74 +18,110 @@ limitations under the License.
...
+    void execute(const Command::Peer::Add& );
+    void execute(const Command::Peer::Remove& );
+    void execute(const Command::Peer::Activate& );
+    void execute(const Command::Peer::Stop& );
+    void execute(const Command::Peer::ChangeRole& );
+
+    namespace detail {
+      void issueStop(std::string ip, Peer& stop_peer);
+
+      void issueActivate(std::string ip, Peer& activate_peer);
```
Comment (takemiyamakoto) [inline irohad/peer_service/peer_service.hpp]: *const*を使えばいいかも。

File: irohad/consensus/sumeragi.cpp (hunk 1 above)
Comment (satellitex) [inline irohad/consensus/sumeragi.cpp]: sumeragi は Bogdan のものにすべて書き換わるので問題ないです。

File: irohad/peer_service/peer_service.hpp [hunk 5]
```diff
@@ -18,73 +18,104 @@ limitations under the License.
...
 
 #include <algorithm>
-#include <datetime/time.hpp>
+#include <cmath>
 #include <memory>
 #include <string>
 #include <vector>
-#include <cmath>
+
+#include <commands.pb.h>
```
Comment (lebdron) [inline irohad/peer_service/peer_service.hpp]: Protobuf is used only as network transport. `model` should be used in this file

**Pull request merged**